src/main/java/ch/wisv/areafiftylan/users/controller/UserProfileRestController.java

```diff
@@ -60,16 +66,28 @@
     }
 
     /**
-     * Add a profile to the current user. An empty profile is created when a user is created, so this method fills the
-     * existing fields
+     * Add a profile to the current user. An empty profile is created when a user is created, so
+     * this method fills the existing fields
      *
      * @param input A representation of the profile
-     *
      * @return The user with the new profile
      */
     @PreAuthorize("isAuthenticated()")
     @PostMapping("/current/profile")
     public ResponseEntity<?> addProfile(@AuthenticationPrincipal User user, @Validated @RequestBody ProfileDTO input) {
+        Collection<RFIDLink> linkCollection = rfidLinkRepository.findRFIDLinksByEmail(user.getEmail());
+        if (linkCollection.isEmpty())
+            return this.addProfile(user.getId(), input);
```

Comment:
{ } please!

Reply:
Also let's add some comments here why this is done.

Reply:
Added!